### DISCUSSION TITLE
Why is the real part of the Sionna phase in the CIR method always zero ?

### DISCUSSION THREAD

**Discussion by claudio966** [NVlabs/sionna#332] [Q&A]
Hi,

I was taking a look at in how Sionna models the phase of a given path in the CIR method, and I didn't understand why all phases only have the imaginary part and the real part is always zero, as shown in the code below taken from ![here](https://github.com/NVlabs/sionna/blob/7eb04f43356333386b4361361188e3810620a0d8/sionna/rt/paths.py#L643). Is there a specific reason for this ?

```python
phase = tf.complex(tf.zeros_like(tau_),
                           -2*PI*self._scene.frequency*tau_)
```

Thank you in advance for your response.

Kind regards,
Cláudio

**Comment by faycalaa** [comment]
Hi,

The line you are referring to computes the exponent of the phase factor.
Phase shift is applied [a few lines below](https://github.com/NVlabs/sionna/blob/7eb04f43356333386b4361361188e3810620a0d8/sionna/rt/paths.py#L648).

**Comment by claudio966** [comment]
I see, now it makes more sense, thanks for the answer.
